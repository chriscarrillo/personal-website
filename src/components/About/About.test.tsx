import {describe, expect, it} from '@jest/globals'
import {render} from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'
import {About} from '.'

describe('<About />', () => {
  it('should render as expected', () => {
    const tree = renderer.create(<About />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show title', async () => {
    const {findByText} = render(<About />)
    await expect(findByText('About')).toBeTruthy()
  })

  it('should show image', async () => {
    const {findByAltText, findByTestId} = render(<About />)
    await expect(findByTestId('chrisPhoto')).toBeTruthy()
    await expect(findByAltText('Chris Carrillo')).toBeTruthy()
  })
})
