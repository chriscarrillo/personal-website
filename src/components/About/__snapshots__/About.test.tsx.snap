// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render as expected 1`] = `
<section
  className="py-4"
>
  <div
    className="d-flex flex-column align-items-center justify-content-center container"
  >
    <h1
      className="text-dark text-uppercase family-semi mb-0"
      data-aos="flip-up"
    >
      About
    </h1>
    <div
      className="d-flex w-100 align-items-center justify-content-center"
    >
      <div
        className="bg-dark mb-2 mr-1 divider"
      />
      <svg
        className="text-dark mb-2"
        fill="currentColor"
        height={40}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width={40}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M5.854 4.146a.5.5 0 010 .708L2.707 8l3.147 3.146a.5.5 0 01-.708.708l-3.5-3.5a.5.5 0 010-.708l3.5-3.5a.5.5 0 01.708 0zm4.292 0a.5.5 0 000 .708L13.293 8l-3.147 3.146a.5.5 0 00.708.708l3.5-3.5a.5.5 0 000-.708l-3.5-3.5a.5.5 0 00-.708 0z"
          fillRule="evenodd"
        />
      </svg>
      <div
        className="bg-dark mb-2 ml-1 divider"
      />
    </div>
    <div
      className="align-items-center row"
    >
      <div
        className="ud-flex justify-content-center col-md-6 col-sm-12"
        data-aos="fade-right"
      >
        <img
          alt="Chris Carrillo"
          className="picture img-fluid rounded"
          data-testid="chrisPhoto"
          src="chris.jpg"
        />
      </div>
      <div
        className="pt-2 col-md-6 col-sm-12"
        data-aos="fade-left"
      >
        <p>
          I'm Chris, a software engineer based in Dallas, Texas.
        </p>
        <p>
          I am a coffee lover and enneagram enthusiast that is passionate about the ins and outs of applications that you see on the web. That includes front-end libraries and frameworks, back-end API's, and relational and non-relational databases.
        </p>
        <p>
          After graduating from California Baptist University with a Bachelor of Science in Software Engineering, I moved to Dallas and started my career as a software engineer, creating intentional and meaningful applications.
        </p>
        <p>
          Just a few of my favorite technologies:
        </p>
        <div
          className="row"
        >
          <div
            className="pl-0 col-md-4 col-sm-12"
          >
            <ul>
              <li>
                React
              </li>
              <li>
                Angular
              </li>
              <li>
                Gatsby
              </li>
            </ul>
          </div>
          <div
            className="pl-0 col-md-4 col-sm-12"
          >
            <ul>
              <li>
                Commerce.js
              </li>
              <li>
                C#
              </li>
              <li>
                Java
              </li>
            </ul>
          </div>
          <div
            className="pl-0 col-md-4 col-sm-12"
          >
            <ul>
              <li>
                SQL Server
              </li>
              <li>
                Firebase
              </li>
              <li>
                Azure
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
